engine/access/rest/middleware/metrics.go

```diff
@@ -11,12 +11,17 @@ import (
 	"github.com/onflow/flow-go/module"
 )
 
-func MetricsMiddleware(restCollector module.RestMetrics) mux.MiddlewareFunc {
+func MetricsMiddleware(restCollector module.RestMetrics, urlToRoute func(string) (string, error)) mux.MiddlewareFunc {
 	metricsMiddleware := middleware.New(middleware.Config{Recorder: restCollector})
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			routeName, err := urlToRoute(req.URL.Path)
+			if err != nil {
+				return
```

Comment:
looks like there are a few REST unittests failing with empty responses. I'm guessing it's related to this return since it effectively drops the request. Can you take a look?

most likely, the tests just need to be updated to use more valid urls/inputs, but not 100% sure.

Reply:
yea, that sounds good. Let's go with `unknown`